znflow/graph.py

```diff
@@ -142,13 +142,56 @@ def get_sorted_nodes(self):
             all_pipelines += nx.dfs_postorder_nodes(reverse, stage)
         return list(dict.fromkeys(all_pipelines))  # remove duplicates but keep order
 
-    def run(self):
-        for node_uuid in self.get_sorted_nodes():
-            node = self.nodes[node_uuid]["value"]
-            if not node._external_:
-                # update connectors
-                self._update_node_attributes(node, handler.UpdateConnectors())
-                node.run()
+    def run(
+        self,
+        nodes: typing.Optional[typing.List[NodeBaseMixin]] = None,
+        immutable_nodes: bool = True,
```

Comment:
Is the `immutable_nodes: bool = True` default value an issue for ZnTrack experiments?